### ISSUE TITLE
for large data >3000 the select dropdown data taking a long time to response

### ISSUE THREAD

**Issue by beasaree** [optimistex/ngx-select-ex#99]
for large data the multiple select dropdown is taking long time more than 10 s.  it will create performance issue . Please can u suggest how can we overcome this issue.

**Comment by DaLiangLee** [comment]
yes same here.. I'm using it for Angular4.....
Now my customer is giving this feedback, which makes me very depressed.

**Comment by optimistex** [comment]
@DaLiangLee 
I can suggest two ways:
- try to load not every data but part of it and use the search. By search event you can filter the data and put less but actual data.
- try the component https://github.com/ng-select/ng-select
it has a virtual scroll that may solve your issue: https://ng-select.github.io/ng-select#/virtual-scroll

**Comment by DaLiangLee** [comment]
Thank you very much for your advice, but what the customer needs is to display all the data and then filter it.
I guess the rendering data is slower...
When I click on the Select box with more data, the response time will exceed 3s.

[https://github.com/ng-select/ng-select#versions ](url)
Not applicable to Angular4, because the project is online and I don't want to make a big upgrade... you know...

In short, thank you very much.

**Comment by casper5822** [comment]
Same for me. Very slow on a 6000 items. Can you do something to improve it ? It's important for our customers.
Thank you

**Comment by optimistex** [comment]
@casper5822 Check the comment above https://github.com/optimistex/ngx-select-ex/issues/99#issuecomment-455502975

I do not have enough time to embed a virtual scroll to this component. If you find a time and implement it, make a pull request then.
